File: TelerikUIExamplesInSwift/TelerikUIExamplesInSwift/StackedAreaChart.swift
//
//  StackedAreaChart.swift
//  Swift Examples
//
//  Copyright (c) 2014 Telerik. All rights reserved.
//

import Foundation

class StackedAreaChart: TKExamplesExampleViewController {
    
    let chart = TKChart()
    
    override init(nibName nibNameOrNil: String?, bundle nibBundleOrNil: NSBundle?) {
        super.init(nibName: nibNameOrNil, bundle: nibBundleOrNil)

        self.addOption("Stacked", action: reloadData)
        self.addOption("Stack 100", action: reloadData)
        self.addOption("No Stacking", action: reloadData)
    }

    required init(coder aDecoder: NSCoder) {
        fatalError("init(coder:) has not been implemented")
    }
    
    override func viewDidLoad() {
        super.viewDidLoad()
        
        chart.frame = self.view.bounds
        chart.autoresizingMask = UIViewAutoresizing(rawValue: UIViewAutoresizing.FlexibleWidth.rawValue | UIViewAutoresizing.FlexibleHeight.rawValue)
        self.view.addSubview(chart)
        
        reloadData()
    }
    
    func reloadData() {
        chart.removeAllData()
        
        var stackInfo: TKChartStackInfo?
        if(self.selectedOption == 0) {
            // >> chart-stack-area-swift
            stackInfo = TKChartStackInfo(ID: 1, withStackMode: TKChartStackMode.Stack)
            // << chart-stack-area-swift
        }
        else if (self.selectedOption == 1)
        {
            // >> chart-stack-area-100-swift
            stackInfo = TKChartStackInfo(ID: 1, withStackMode: TKChartStackMode.Stack100)
            // << chart-stack-area-100-swift
        }
        
        // >> chart-stack-area-swift
        for _ in 0..<3 {
            var array = [TKChartDataPoint]()
            for i in 0..<8 {
                array.append(TKChartDataPoint(x: i+1, y: Int(arc4random() % (100))))
            }
            let series = TKChartAreaSeries(items: array)
            series.selection = TKChartSeriesSelection.Series
            series.stackInfo = stackInfo
            chart.addSeries(series)
        }
        // << chart-stack-area-swift

        chart.reloadData()
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
}

File: Examples/Examples/Docs/StackedCharts/StackedAreaChart.cs
using System;
using System.Collections.Generic;

using Foundation;
using UIKit;

using TelerikUI;

namespace Examples
{
	[Register("StackedAreaChart")]
	public class StackedAreaChart: XamarinExampleViewController
	{
		TKChart chart;

		public override void ViewDidLoad ()
		{
			this.AddOption ("Stacked", reloadData);
			this.AddOption ("Stacked 100", reloadData);
			this.AddOption ("No Stacking", reloadData);

			base.ViewDidLoad ();

			chart = new TKChart (this.View.Bounds);
			chart.AutoresizingMask = UIViewAutoresizing.FlexibleWidth | UIViewAutoresizing.FlexibleHeight;
			this.View.AddSubview (chart);

			this.reloadData ();
		}

		void reloadData()
		{
			chart.RemoveAllData ();

			TKChartStackInfo stackInfo = null;

			if (this.SelectedOption == 0) 
			// >> chart-stack-area-cs
				stackInfo = new TKChartStackInfo (new NSNumber (1), TKChartStackMode.Stack);
			// << chart-stack-area-cs
			else if (this.SelectedOption == 1)
			// >> chart-stack-area-100-cs
				stackInfo = new TKChartStackInfo (new NSNumber (1), TKChartStackMode.Stack100);
			// << chart-stack-area-100-cs

			// >> chart-stack-area-cs
			Random r = new Random ();
			for (int i = 0; i < 3; i++) {
				List<TKChartDataPoint> list = new List<TKChartDataPoint> ();
				for (int j = 0; j < 8; j++) {
					list.Add(new TKChartDataPoint(new NSNumber(j+1), new NSNumber(r.Next() % 100)));
				}
				TKChartAreaSeries series = new TKChartAreaSeries (list.ToArray());
				series.SelectionMode = TKChartSeriesSelectionMode.Series;
				series.StackInfo = stackInfo;
				chart.AddSeries (series);
			}
			// << chart-stack-area-cs
		}
	}
}

